File: tests/fixtures/sample.c
/* A sample C file for testing TODO detection */

#include <stdio.h>

int main() {
    // TODO: Add command line argument parsing
    printf("Hello\n");

    /* HACK: Buffer overflow workaround
     * This needs a proper fix
     */
    char buf[256];

    return 0;
}

File: tests/fixtures/sample.rs
// A sample Rust file for testing TODO detection

fn main() {
    // TODO: Add proper argument parsing
    let x = 42;

    // FIXME(alice): This calculation is wrong
    let y = x * 2;

    // HACK: Temporary workaround for issue #789
    println!("{}", y);

    let todo_string = "This has TODO in a string but it's not a comment";
}

// TODO(bob, #123, p:high): Implement error handling
fn placeholder() {}
